aijson/utils/action_utils.py

```diff
@@ -260,82 +262,112 @@ def build_action_description(
-                    base_description=base_description,
-                    base_markdown_description=base_markdown_description,
-                    field=field,
-                    include_paths=include_paths,
-                    name=output_attr,
-                    alias_name=action_id,
+            if is_basemodel_subtype(outputs_type):
+                if issubclass(outputs_type, DefaultOutputOutputs):
+                    output_attr = outputs_type._default_output
+                    field = outputs_type.model_fields[output_attr]
+                    annotated_field = _build_annotated_field(
+                        base_description=base_description,
+                        base_markdown_description=base_markdown_description,
+                        field=field,
+                        include_paths=include_paths,
+                        name=output_attr,
+                        alias_name=action_id,
+                    )
+                    union_elements.append(annotated_field)
+                union_elements.extend(
+                    _get_recursive_subfields(
+                        outputs_type,
+                        base_description,
+                        base_markdown_description,
+                        include_paths=include_paths,
+                        name_prefix=f"{action_id}.",
+                    )
                 )
-                union_elements.append(annotated_field)
-            union_elements.extend(
-                _get_recursive_subfields(
-                    outputs_type,
-                    base_description,
-                    base_markdown_description,
-                    include_paths=include_paths,
-                    name_prefix=f"{action_id}.",
-                )
-            )
-        else:
-            union_elements.append(
-                _build_annotated_field(
-                    base_description=base_description,
-                    base_markdown_description=base_markdown_description,
-                    field=FieldInfo(
-                        annotation=outputs_type,
-                    ),
-                    include_paths=include_paths,
-                    name=action_id,
+            else:
+                union_elements.append(
+                    _build_annotated_field(
+                        base_description=base_description,
+                        base_markdown_description=base_markdown_description,
+                        field=FieldInfo(
+                            annotation=outputs_type,
+                        ),
+                        include_paths=include_paths,
+                        name=action_id,
+                    )
                 )
-            )
 
-    if union_elements:
-        return Union[tuple(union_elements)]  # type: ignore
-    return str
+        if union_elements:
+            return Union[tuple(union_elements)]  # type: ignore
+
+    links = build(action_config)
+    return (links, loop_elements)  # type: ignore
```

Comment:
```suggestion
    return links, loop_elements
```

what should the return annotation of this function be, to get rid of the type ignore?